api4/post.go

```diff
@@ -405,6 +406,53 @@ func updatePost(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(c.App.PostWithProxyAddedToImageURLs(rpost).ToJson()))
 }
 
+func rethreadPost(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequirePostId()
+	if c.Err != nil {
+		return
+	}
+
+	if !*c.App.Config().ServiceSettings.ExperimentalRethreadMessages {
+		c.SetPermissionError(model.PERMISSION_EDIT_POST)
```

Comment:
I think an AppError wrapping `http.StatusNotImplemented` would be better suited here.

Reply:
Agreed. A few of these do involve incorrect permissions, but the other cases should be returning errors that more accurately explain the problem. You can see some examples of that [here](https://github.com/mattermost/mattermost-server/blob/master/api4/emoji.go#L36)